scraper/mapharma.py

```diff
@@ -0,0 +1,160 @@
+    return reasons
+
+def getProfile(url, client: httpx.Client = DEFAULT_CLIENT):
+    profile = {}
+    profile['reasons'] = []
+    try:
+        r = client.get(url)
+        r.raise_for_status()
+        soup = BeautifulSoup(r.content, 'html.parser')
+        reasons = getReasons(soup)
+        if reasons != []:
+            profile['reasons'] = reasons
+        return profile
+    except httpx.HTTPStatusError:
+        return profile
+
+def getProfiles(zip, client: httpx.Client = DEFAULT_CLIENT):
+    index = 0
+    loop = True
+    result = []
+    while True:
+        base_url = f"https://mapharma.net/{zip}-{index}" if index > 0 else f'https://mapharma.net/{zip}'
+        try:
+            r = client.get(base_url)
+            r.raise_for_status()
+            soup = BeautifulSoup(r.content, 'html.parser')
+            reasons = getReasons(soup)
+            name = getName(soup)
+            address = getAddress(soup)
+            payload = { 'id': index, 'url': base_url, 'zip': zip, 'id': index, 'name': name, 'address': address, 'reasons': reasons}
+            result.append(payload)
+            index += 1
+        except httpx.HTTPStatusError:
+            if index == 0:
+                index = 1
+            else:
+                return result
+
+def parseAllZip():
+    profiles = dict()
+    with open("data/input/codepostal_to_insee.json", "r") as json_file:
+        zips = json.load(json_file)
+        for zip in zips.keys():
+            logger.info(f'Searching cp {zip}...')
+            for profile in getProfiles(zip, DEFAULT_CLIENT):
+                if zip not in profiles:
+                    profiles[zip] = [] 
+                profiles[zip].append(profile)
+            if zip in profiles and len(profiles[zip]) > 0:
+                logger.info(f'found {len(profiles[zip])} in cp {zip}')
+    with open("data/input/mapharma.json", "w") as json_file:
+        json.dump(profiles, json_file, indent = 4)
+
+def bruteForce():
+    profiles = dict()
+    with open("data/input/codepostal_to_insee.json", "r") as json_file:
+        zips = json.load(json_file)
+        for zip in zips.keys():
+            for profile in getProfiles(zip):
+                if zip not in profiles:
+                    profiles[zip] = [] 
+                profiles[zip].append(profile)
+    with open("data/input/mapharma.json", "w") as json_file:
+        json.dump(profiles, json_file, indent = 4)
+
+def getSlots(campagneId, optionId, start_date, client: httpx.Client = DEFAULT_CLIENT):
+    #logger.debug((f'campagneId: {campagneId}, optionId: {optionId}, start_date: {start_date}')
+    base_url = f'https://mapharma.net/api/public/calendar/{campagneId}/{start_date}/{optionId}'
+    client.headers.update({'referer': 'https://mapharma.net/'})
+    r = client.get(base_url)
+    r.raise_for_status()
+    return r.json()
+
+def parseSlots(slots):
+    first_availability = None
+    for key in slots.keys():
```

Comment:
Vu qu'on va utiliser la clé et la valeur :
```suggestion
    for key, day in slots.items():
```

Reply:
bien vu, je remplace par for date, day_slots in slots.items():